zanata-client-commands/src/main/java/org/zanata/client/commands/push/RawPushCommand.java

```diff
@@ -75,10 +94,164 @@ public RawPushCommand(PushOptions opts, RestClientFactory clientFactory) {
+     * - Source file extension that is not supported from file type will be ignored.
+     *
+     * eg. input pattern --file-types properties_utf8,properties[txt],plain_text[md;markdown],html[txt],idml
+     *
+     * @param acceptedTypes
+     * @param inputFileTypes
+     */
+    private Map<DocumentType, Set<String>> validateFileTypes(
+            List<DocumentType> acceptedTypes, List<String> inputFileTypes) {
+
+        Map<DocumentType, Set<String>> filteredFileTypes = new HashMap<>();
+
+        for (String typeWithExtension : inputFileTypes) {
+            String fileType = extractFileType(typeWithExtension);
+            List<String> extensions = extractExtensions(typeWithExtension);
+
+            validateInputFileType(fileType, extensions, acceptedTypes);
+
+            DocumentType inputFileType = DocumentType.getByName(fileType);
+
+            //skip file type if its not supported
+            if (inputFileType == null) {
+                String msg =
+                        "Requested file type '"
+                                + fileType
+                                + "' is not supported by the target server and will be ignored.";
+                consoleInteractor.printfln(DisplayMode.Warning, msg);
+                continue;
+            }
+
+            //throw error if file type is input more than once
+            if (filteredFileTypes.containsKey(inputFileType)) {
+                String msg =
+                    "File type '" + fileType + "' cannot be used more than once in option --file-types.";
+                log.error(msg);
+                throw new RuntimeException(msg);
+            }
+
+            Set<String> filteredExtensions = new HashSet<>(extensions);
+
+            if (!extensions.isEmpty()) {
+                //throw exception if user input extension if not part of DocumentType supported source extension
+                for (String extension: extensions) {
+                    if (!inputFileType.getSourceExtensions()
+                            .contains(extension)) {
+                        String msg =
+                                "Extension '"
+                                        + extension
+                                        + "' for file type '"
+                                        + fileType
+                                        + "' is not supported by the target server and will be ignored.";
+                        consoleInteractor.printfln(DisplayMode.Warning, msg);
+                        filteredExtensions.remove(extension);
+                    }
+                }
+            }
+
+            /**
+             * Use the extensions from typeWithExtension input if exists,
+             * otherwise, use the extensions from server.
+             */
+            filteredExtensions =
+                filteredExtensions.isEmpty() ? inputFileType.getSourceExtensions()
+                            : filteredExtensions;
```

Comment:
I don't think we should be running the command using default extensions unless they actually asked for them (by not including any extensions). If something they put in is not valid, I think it is better to just fail the command and tell them about it so they can fix it, so they have a chance to make it work the way they intended.